sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts

```diff
@@ -108,10 +113,38 @@ export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
         "Not all required fields to update a received MSFAA record were provided.",
       );
     }
-
+    // Check if the msfaa number needs to be reactivated or simply updated with a signed date.
+    const retrievedMSFAARecord = await this.repo.findOne({
+      select: {
+        id: true,
+        studentId: true,
+        referenceApplicationId: true,
+        cancelledDate: true,
+        dateSigned: true,
+        offeringIntensity: true,
+      },
+      where: {
+        msfaaNumber,
+      },
+    });
+    if (
+      !retrievedMSFAARecord.dateSigned &&
+      retrievedMSFAARecord.cancelledDate
+    ) {
```

Comment:
Can we please have the if removed and move it to the typeorm query, as findone like below? As findOne returns either an entity or null

![image](https://github.com/bcgov/SIMS/assets/62901416/06a9a701-3eb0-4373-bfed-793b5e486eb9)

Reply:
I believe the where condition should be as below:

`
where: {
    msfaaNumber,
    offeringIntensity,
    cancelledDate: Not(IsNull()),
}

Reply:
I took it from your previous example, But you got the idea right?

Reply:
I took it from your previous example, But you got the idea right?